src/sync/lookupApiUtils.js

```diff
@@ -120,8 +135,26 @@ const processInsuranceResponse = response =>
     })
   );
 
-export const processPatientResponse = response =>
-  response.map(
+const processResponse = response => {
+  const { ok, status, url, headers, json } = response;
+  if (ok) {
+    const { error: responseError } = json;
+    if (responseError) throw new BugsnagError(responseError, { url, headers });
+    if (!json.length) throw new Error(ERROR_CODES.EMPTY_RESPONSE);
+    return json;
+  }
+  switch (status) {
+    case 400:
+    default:
+      throw new Error(ERROR_CODES.CONNECTION_FAILURE);
+    case 401:
+      throw new BugsnagError(ERROR_CODES.INVALID_PASSWORD, { url, headers });
```

Comment:
Do you think putting this switch into the error handler passed into the hook could be a good use? Rather than using the response handler for both error logic and ok logic - I always find error handling annoying 🤣 (no change just interested!)

Reply:
I thought about this a bit and did think about it, but since this is a switch on a non-error response, it doesn't really make sense to put it in an error handler (no errors have been thrown yet). 

On revisiting it, I think a possible idea would be to have a custom `HTTPError` and do something like:

```
const processResponse = response => {
  const { ok, status, ... } = response;
  if (!ok) throw HTTPError(status);
  ...
}

const processError = error => {
  if (error instanceOf HTTPError) {
    switch (error.status) {
       ...
    }
  }
}
```